lib/_http_client.js

```diff
@@ -569,10 +569,20 @@ ClientRequest.prototype.setTimeout = function(msecs, callback) {
 };
 
 ClientRequest.prototype.setNoDelay = function() {
-  this._deferToConnect('setNoDelay', arguments);
+  var argsLength = arguments.length;
+  var args = new Array(argsLength);
+  for (var i = 0; i < argsLength; i++) {
+    args[i] = arguments[i];
+  }
```

Comment:
I'm not sure about backwards vs. forwards, but generally I've seen that for loops are faster than while loops in V8. Take a look at this benchmark I just ran (I put some of my results in a comment at the bottom): https://gist.github.com/woollybogger/7364d4cf4599a1fc0ae7